app/Models/Chat/Channel.php

```diff
@@ -20,40 +20,142 @@
 
 namespace App\Models\Chat;
 
-use App\Interfaces\Messageable;
+use App\Exceptions\API;
 use App\Models\User;
+use Carbon\Carbon;
 
-class Channel extends Model implements Messageable
+class Channel extends Model
 {
     protected $primaryKey = 'channel_id';
     protected $dates = [
         'creation_time',
     ];
 
+    const TYPES = [
+        'public' => 'PUBLIC',
+        'private' => 'PRIVATE',
+        'multiplayer' => 'MULTIPLAYER',
+        'spectator' => 'SPECTATOR',
+        'temporary' => 'TEMPORARY',
+        'pm' => 'PM',
+        'group' => 'GROUP',
+    ];
+
     public function messages()
     {
-        $this->hasMany(Message::class, 'channel_id');
+        $messages = $this->hasMany(Message::class, 'channel_id');
+
+        if ($this->type === self::TYPES['public']) {
```

Comment:
This doesn't work with `::with()`, so 
```
Channel::where('type', 'PUBLIC')->with('messages')->get()
```
doesn't apply the where clause, so I'm beginning to think we shouldn't be baking in extra behaviour into association methods.